src/arguments.jl

```diff
@@ -0,0 +1,36 @@
+"""
+    Arguments{names,T}
+
+    Arguments(; kw...)
+    Arguments(nt::NamedTuple)
+
+A wrapper for a `NamedTuple`.
+"""
+struct Arguments{names,T}
+    nt::NamedTuple{names,T}
+end
+
+Arguments(; kw...) = Arguments(values(kw))
+Arguments{names}(vals) where {names} = Arguments(NamedTuple{names}(vals))
+
+nt(a::Arguments) = getfield(a, :nt)
+
+for f in (:length, :firstindex, :lastindex, :isempty, :keys, :values)
+    @eval Base.$f(a::Arguments) = $f(nt(a))
+end
+
+for f in (:(==), :isequal, :<, :isless, :isapprox)
+    @eval Base.$f(a1::Arguments, a2::Arguments) = $f(nt(a1), nt(a2))
+end
+
+Base.iterate(a::Arguments, iter=1) = iterate(nt(a), iter)
```

Comment:
```suggestion
Base.iterate(a::Arguments, iter...) = iterate(nt(a), iter...)
```
The default argument isn't needed

Reply:
okay